api/src/main/java/org/openmrs/module/fhir/api/util/FHIRPractitionerUtil.java

```diff
@@ -139,7 +139,39 @@ public static Practitioner generatePractitioner(Provider provider) {
 		return practitioner;
 	}
 	
-	public static Person generateOpenMRSPerson(Practitioner practitioner) {
+	public static Person generateOpenMRSPerson(Person personFromRequest) {
+		Set<PersonName> nameList = personFromRequest.getNames();
+		PersonName usedName = null;
+		for (PersonName prsnName : nameList) {
+			if (prsnName.isPreferred()) {
+				usedName = prsnName;
+				break;
+			}
+		}
+		
+		Set<Person> personList = Context.getPersonService().getSimilarPeople(usedName.getFullName(),
+		    1900 + personFromRequest.getBirthdate().getYear(), personFromRequest.getGender()); // filter Persons
+		boolean createPerson = false;
+		if (personList.isEmpty()) {
+			createPerson = true;
+		} else {
+			if (personList.size() != 1) {
+				createPerson = true;
+			}
+		}
+		
+		Person personForProvider = null;
+		if (createPerson) {
+			personForProvider = Context.getPersonService().savePerson(personFromRequest);
+		} else {
+			for (Iterator<Person> pers = personList.iterator(); pers.hasNext();) {
+				personForProvider = pers.next();
```

Comment:
What is actully ddoing here? Iterating all the loop and person for provider will be the last ?